sdk/src/test/java/tech/dojo/pay/sdk/payemntintent/PaymentIntentProviderTest.kt

```diff
@@ -0,0 +1,246 @@
+package tech.dojo.pay.sdk.payemntintent
+
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import tech.dojo.pay.sdk.DojoPaymentIntentResult
+import tech.dojo.pay.sdk.payemntintent.data.PaymentIntentRepository
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class PaymentIntentProviderTest {
+
+    private val paymentIntentRepository: PaymentIntentRepository = mockk()
+    private val repositoryScope: CoroutineScope = CoroutineScope(UnconfinedTestDispatcher())
+
+    private val paymentIntentProvider = PaymentIntentProvider(
+        paymentIntentRepository,
+        repositoryScope,
+    )
+
+    @Test
+    fun `when fetchPaymentIntent success onPaymentIntentSuccess should be invoked`() = runTest {
```

Comment:
the test methods name dont match the result that are return
```suggestion
    fun `when fetchPaymentIntent success onPaymentIntentSuccess should return service json`() = runTest {
```